### ISSUE TITLE
Toolbar Options doesn't rerender after MUI version 2.0.0

### ISSUE THREAD

**Issue by iKrushYou** [gregnb/mui-datatables#702]
Back in version 2.0.0, I was able to pass through a prop and have it rerender an element in the toolbar [for example](https://codesandbox.io/s/mui-datatable-issue-i6drf) I have a refresh button with a spinner that shows it's refreshing when you click it. If you change the version here to anything later than 2.0.0, the functionality breaks.

I feel kind of awkward that this issue is so short, but all the required info can be found in the codesandbox. 

Basically, I had a button that would conditionally display a spinner inside of the toolbar and now the toolbar doesn't re-render the button when updating the prop.

**Comment by gabrielliwerant** [comment]
Be careful with using unsupported functionality. You're using the latest and greatest hooks from react, but this library is built on an earlier version of react. If I reconstruct the example using classes, it works for me: https://codesandbox.io/embed/mui-datatable-issue-qj9us.

**Comment by iKrushYou** [comment]
Ah.. damn I didn't realize that hooks weren't supported yet. I wonder what the underlying reason is for this issue though that causes it to work fine on an older version

**Comment by gabrielliwerant** [comment]
@iKrushYou Older versions were forcibly updating derived state from props using deprecated methods. Doing so had to be removed in order to open up an upgrade path, but were also causing other subtle bugs.
